### DISCUSSION TITLE
Sweethome 3D SVG Export

### DISCUSSION THREAD

**Discussion by tsx-5** [ExperienceLovelace/ha-floorplan#405] [Q&A] [answered]
Hi All,

New to this project, excited to give it a whirl.  I've used SH3D to create a pretty good model of my home (3 floors) and note that it can export out to SVG.  The YouTube video points to using Inkscape to manage / create SVG file.  So, my first question is - do I actually need Inkscape, or will the SVG output from SH3D work?

Regardless of answer #1, I've been able to open the SVG file from SH3D in Inkscape - which shows a 2D view of a floor with multiple (like a whole list) of layers.  The names of these layers is not very easy to know what is what.  So, 2nd question, is there a way in SH3D to name the layers coming out of it? 

Thanks in advance,

**Comment by OzGav** [comment]
I expect the SVG will load but the challenge will be controlling the elements to animate your house. I actually rendered a PNG from SH3D and imported that into Inkscape. Inkscape is not mandatory but it provides an easy mechanism to create layers, add elements to those layers and then name them so they can be manipulated by ha-floorplan.

**Reply by tsx-5** [reply]
Thanks for the response.  For some reason I was thinking the SVG file from SH3D would be the same as from Inkscape.  Realizing that the SVG from SH3D is only 2D and doesn't have the rendering as part of it.  That's where Inkscape comes in.  Think I've got a bit better of an understanding now.

**Comment by Trzinka** [comment]
Hi.
**As OzGav has already written, I am still completing it.**
### First of all, if you export from SH3D to an SVG file without using the create photo function, you will get this image quality:

![SH3D-Not render](https://github.com/ExperienceLovelace/ha-floorplan/assets/40424965/6e64469b-9ed8-4fb8-ad57-20d21405170b)

#  This is the quality of the image according to your method

## If you use this method:

![SH3D](https://github.com/ExperienceLovelace/ha-floorplan/assets/40424965/77e56990-fafc-4f2f-9ad2-942c4e766b4b)

You can expect image quality something like this:

![SH3D-Render](https://github.com/ExperienceLovelace/ha-floorplan/assets/40424965/b47d5f15-268e-45d5-b9dc-be46c857c8a3)

### As you can see, the difference is huge!!!

**But it is true that if you want a good quality of the finished product, you will spend a lot of time.**

### Believe me, it's worth it!!!

**Reply by tsx-5** [reply]
When I exported from SH3D, I didn't even get the colors... I only got the black/white floorplan :-/  That said, the 3D render does look WAY better.

**Reply by OzGav** [reply]
Must be a setting or something you are selecting. I actually used this to render but it may be redundant now I am not sure http://www.sweethome3d.com/blog/2020/01/31/yafaray_rendering_plug_in.html

**Reply by Trzinka** [reply]
### I use 2 monitors! Don't bother (If you use this method:)!

# This is what it looks like on one screen:

![SH3D-one screen](https://github.com/ExperienceLovelace/ha-floorplan/assets/40424965/aeb8f1ea-e35d-44d7-83dd-f5382c4bfe1a)

**In the lower right window, click with the right mouse button and you will get a screen as shown in the second picture.**

_And it will make you a color image that you can save as a PNG. The issue may take some time before the render is finished. It depends on the power of the processor._

**Reply by Trzinka** [reply]
I don't want to confuse you, I'm just giving you this as a suggestion: 

https://github.com/adizanni/floor3d-card

The quality of the final product is worse and requires a lot more css settings.
But if you want something quick because you have already SH3D made it, it might be worth a try!!!

### To be honest, I tried it myself and was disappointed with the picture quality!

**Reply by tsx-5** [reply]
Not confusing me 😃.  Appreciate all the info, will be sticking with your suggested way forward, as you noted it looks the best.

**Reply by Trzinka** [reply, marked as answer]
If you are satisfied with the answer, click on the button
 ```Mark as answer``` 
an then button 
```Close discusion```!!! 
**You can always open a new discussion if you need anything else.**
